nexus/src/app/saga.rs

```diff
@@ -79,12 +77,28 @@ use uuid::Uuid;
 pub(crate) fn create_saga_dag<N: NexusSaga>(
     params: N::Params,
 ) -> Result<SagaDag, Error> {
-    let builder = DagBuilder::new(SagaName::new(N::NAME));
-    let dag = N::make_saga_dag(&params, builder)?;
-    let params = serde_json::to_value(&params).map_err(|e| {
-        SagaInitError::SerializeError(String::from("saga params"), e)
-    })?;
-    Ok(SagaDag::new(dag, params))
+    N::prepare(&params)
+}
+
+/// Interface for kicking off sagas
+///
+/// See [`SagaExecutor`] for the implementation within Nexus.  Some tests use
+/// alternate implementations that don't actually run the sagas.
+pub(crate) trait SagaStarter: Send + Sync {
```

Comment:
this is super pedantic, please feel free to ignore me, but I feel like the general idiom for trait names is that they're verbs, rather than nouns (when it's grammatically reasonable), e.g. `io::Write`, `iter::Extend`, etc. so, i _might_ have called this 
```suggestion
pub(crate) trait StartSaga: Send + Sync {
```

Take it or leave it, though.